### DISCUSSION TITLE
Import graphQL statements from another file

### DISCUSSION THREAD

**Discussion by Kpizzle** [artilleryio/artillery#1462] [Artillery]
Hi there, 

I've started to look into artillery and create a proof of concept for my QA team to start using it for load testing. 

I've got a basic flow all set up with auth and doing everything I'd like. I'm now looking to see if I meet further requirements that we've set as a QA team. 

Our testing infrastructure is a microservice architecture, with about 30 services, and we have 4 copies of this, then UAT > Staging > Production. 

I'd like to know if there was a way that I could have my graphQL queries/mutations in one file and reference them in the scenario/YAML file. So we keep the queries DRY and we don't have to duplicate code. 

Thanks.

**Comment by hassy** [comment]
hi @Kpizzle 👋 

`config.environments` may be helpful if you're looking to re-use the same scenario in multiple different environments (e.g. between UAT, staging and production):

https://www.artillery.io/docs/guides/guides/test-script-reference#environments---config-profiles

A typical layout of an Artillery test suite is something like:

```
- scenarios/
  - scenario1.yaml
  - scenario2.yaml
- config.yaml
```

Where each file under `scenarios` contains one scenario, and `config.yaml` contains configuration for each environment, as well as plugins, loading custom JS code etc.
